File: apps/motoboy-pwa/src/routeStopTypeLabels.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { routeStopTypeLabel } from "./routeStopTypeLabels";

describe("routeStopTypeLabel", () => {
  it("maps known route stop types to readable labels", () => {
    assert.equal(routeStopTypeLabel("COLETA"), "Coleta");
    assert.equal(routeStopTypeLabel("ENTREGA"), "Entrega");
  });

  it("formats unknown route stop types without raw underscores", () => {
    assert.equal(routeStopTypeLabel("RETIRADA_EXTRA"), "Retirada extra");
  });

  it("uses a safe fallback for blank route stop types", () => {
    assert.equal(routeStopTypeLabel("   "), "Parada");
  });
});

File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/RouteStopTypeLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class RouteStopTypeLabelsTest {
  @Test
  fun `maps known route stop types to readable labels`() {
    assertEquals("Coleta", routeStopTypeLabel("COLETA"))
    assertEquals("Entrega", routeStopTypeLabel("ENTREGA"))
  }

  @Test
  fun `formats unknown route stop types without raw underscores`() {
    assertEquals("Retirada extra", routeStopTypeLabel("RETIRADA_EXTRA"))
  }

  @Test
  fun `uses a safe fallback for blank route stop types`() {
    assertEquals("Parada", routeStopTypeLabel("   "))
  }
}
